zigbee: Refactor logging ZBOSS traces (#5279)
nrfconnect/sdk-nrf: doc/nrf/ug_zigbee_configuring.rst

The stack logs are provided in a binary format and you can configure how they are printed:

* :option:`CONFIG_ZBOSS_TRACE_HEXDUMP_LOGGING` - Stack logs are printed as hexdump using Zephyr's :ref:`zephyr:logging_api`.
  This option is enabled by default.
* :option:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - Stack logs are printed in the binary format using one of the following independent serial backends of your choice:

  * :option:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - UART serial.
    This backend is enabled by default.
  * :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - USB CDC serial.

  To specify the serial device, set :option:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME`.
  By default, this option is set to the following values:

  * ``UART_1`` if :option:`CONFIG_ZBOSS_TRACE_UART_LOGGING` is selected.
  * ``CDC_ACM_1`` if :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` is selected.

* :option:`CONFIG_ZBOSS_TRACE_BINARY_NCP_TRANSPORT_LOGGING` - Stack logs are printed in the binary format using the NCP transport channel.

> Stack logs are stored in the internal buffer if they are printed using Zephyr's :ref:`zephyr:logging_api` or the independent serial backend.

Reviewer: When they are not buffered?
The NCP transport? What about the `static uint8_t uart_tx_buf_mem[CONFIG_ZIGBEE_UART_TX_BUF_LEN];` ?

Author: Yes, they are buffered, just don't use the same buffer. Will update this

* :option:`CONFIG_ZBOSS_TRAF_DUMP` - Enables logging of the received 802.15.4 frames over ZBOSS trace log if :option:`CONFIG_ZBOSS_TRACE_LOG_LEVEL` is set; disabled by default.

The stack logs are provided in a binary format and you can configure how they are printed:

* :option:`CONFIG_ZBOSS_TRACE_HEXDUMP_LOGGING` - Stack logs are printed as hexdump using Zephyr's :ref:`zephyr:logging_api`.
  This option is enabled by default.
* :option:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - Stack logs are printed in the binary format using one of the following independent serial backends of your choice:

  * :option:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - UART serial.
    This backend is enabled by default.
  * :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - USB CDC serial.

  To specify the serial device, set :option:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME`.
  By default, this option is set to the following values:

  * ``UART_1`` if :option:`CONFIG_ZBOSS_TRACE_UART_LOGGING` is selected.
  * ``CDC_ACM_1`` if :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` is selected.

.. note::

>     When :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` is selected, it is required that USB CDC serial is started correctly by the application.

Reviewer: Note: Consider being more specific, e.g. attaching a code snippet or pointing to a one.

* :option:`CONFIG_ZBOSS_TRAF_DUMP` - Enables logging of the received 802.15.4 frames over ZBOSS trace log if :option:`CONFIG_ZBOSS_TRACE_LOG_LEVEL` is set; disabled by default.

The stack logs are provided in a binary format and you can configure how they are printed:

* :option:`CONFIG_ZBOSS_TRACE_HEXDUMP_LOGGING` - Stack logs are printed as hexdump using Zephyr's :ref:`zephyr:logging_api`.
  This option is enabled by default.
* :option:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - Stack logs are printed in the binary format using one of the following independent serial backends of your choice:

  * :option:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - UART serial.
    This backend is enabled by default.
  * :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - USB CDC serial.

  To specify the serial device, set :option:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME`.
  By default, this option is set to the following values:

  * ``UART_1`` if :option:`CONFIG_ZBOSS_TRACE_UART_LOGGING` is selected.
  * ``CDC_ACM_1`` if :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` is selected.

.. note::

>     When :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` is selected, it is required that USB CDC serial is started correctly by the application.

Reviewer 2: ```suggestion
    When you select :option:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING`, make sure that the USB CDC serial is correctly started by the application.
```

>     USB peripheral must be enabled and UART line control signals must be correctly set.

Reviewer 2: ```suggestion
    The application must enable the USB peripheral and set the UART line control signals.
```
Who sets these? The actor is missing.

>     You can see the :c:func:`main` of :ref:`NCP sample <zigbee_ncp_sample>` for how this can be implemented.

Reviewer 2: ```suggestion
   See the :c:func:`main` of :ref:`NCP sample <zigbee_ncp_sample>` for an example of how to implement the USB CDC serial support.
```

>     Because of that, for :ref:`NCP sample <zigbee_ncp_sample>` this step can be skipped.

Reviewer 2: Because of what exactly? This suggests that because you've looked at the NCP sample's main, you can skip this step...

Author: The USB CDC implementation mentioned in previous sentence, NCP sample already implements that.
I'd simplify line 172  to "This can be skipped only for NCP sample but you must correctly configure and start USB CDC Serial for other applications.

> * :option:`CONFIG_ZBOSS_TRACE_BINARY_NCP_TRANSPORT_LOGGING` - Stack logs are printed in the binary format using the NCP transport channel.

Reviewer 2: I doubt this will show up as a continuation of the list because of the missing indent of the entire note in lines 168-172.

Author: Changed and checked locally, works
